src/pages/Scholarships.jsx

```diff
@@ -26,14 +29,55 @@ function ScholarshipsPage() {
   const [error, setError] = useState();
   const [sortField, setSortField] = useState('deadline');
   const [sortDir, setSortDir] = useState('asc');
+  const [amountFilterVals, setAmountFilterVals] = useState({
+    minAmount: 0,
+    maxAmount: 0,
+  });
+  const location = useLocation();
+  const history = useHistory();
+
+  useEffect(() => {
+    const params = queryString.parse(location.search, { parseNumbers: true });
+    const minParam = params.min ?? 0;
+    const maxParam = params.max ?? 0;
+    setAmountFilterVals({
+      minAmount: minParam > 0 ? minParam : 0,
+      maxAmount: maxParam > 0 ? maxParam : 0,
+    });
+
+    const newParams = { ...params };
+    if (!Number.isInteger(minParam) || minParam <= 0) delete newParams.min;
+    if (!Number.isInteger(maxParam) || maxParam <= 0) delete newParams.max;
+    history.push({ search: queryString.stringify(newParams) });
```

Comment:
Is there a way this can be done outside the `useEffect`? Something like
```
params = ...;
if params.max not valid:
  delete params.max
  history.replace with params ("replace" since the param was "unclean")
if params.min not valid:
  delete params.min
  history.replace with params ("replace" since the param was "unclean")
```
That way the other effects and the rest of the component don't load until the params are cleaned.

Reply:
Yeah I got it to work outside of useEffect.